src/lib/auth/README.md

```diff
@@ -0,0 +1,45 @@
+## Authentication Library
+Initialize authentication library (authenticator). Based on the authentication type a strategy
+will be prepared.
+
+**The library is promised based, all exposed methods return promises.**
+
+The module acts like a singleton. Both the strategy and the authentication options are saved in
+the package's object which allows us to only initialize the library once. Meaning that after we
+initialize the library it will be shared throughout the entire codebase without the need of
+re-initialization.
+
+Currently implemented strategies are:
+ - `wpcom-oauth` - regular wpcom oAuth process using `wpcom-oauth-cors` package
+ - `wpcom-oauth-by-token` - regular wpcom oAuth based on given token using `wpcom-xhr-request`
+ - `wpcom-proxy-iframe` - proxy iframe wpcom authentication using Calypso's wpcom object
+
+ ## API
+ `init( settings )` - Initialize authentication library (authenticator). Based on the authentication type a strategy will be selected.
+
+ #### Available settings
+
+| Key | Value | Required | Default | Available Values | Description |
+| --- | --- | --- | --- | --- | --- |
+| `type` | string | Required | - | `wpcom-oauth` \| `wpcom-oauth-by-token` \| `wpcom-proxy-iframe` | A valid implemented strategy type |
+| `options` | object | Optional | - | - | Set of authentication options
+| `options.token` | string | Required by `wpcom-oauth-by-token` | - | - | oauth accessToken
+| `options.proxy` | object | Required by `wpcom-proxy-iframe` | - | - | wpcom auth proxy object
+
+`authorizeChat( state )` - Authorize and start chat session by signing a JWT and returns jwt, userId, skills and geolocation prepared.
```

Comment:
> Authorize and start chat session by signing a JWT and returns jwt, userId, skills and geolocation prepared.

Minor: There are two subjects implied in this sentence. It reads a little awkward: 
1. The consumer. `Authorize and start a chat session` implies: "by calling this function, you autorize and start a chat session..."
2. The function. `returns jwt, userId, skills and geolocation prepared.`. This is the function doing something.

It might be best to separate these clauses, or rewrite with one subject in mind e.g. "Authorizes and starts... and returns..."

Reply:
Great suggestion, updated.